### ISSUE TITLE
CFGFast fails on /bin/ls (nop_length is None)

### ISSUE THREAD

**Issue by lockshaw** [angr/angr#1455]
**Describe the bug**

Running CFGFast on /bin/ls fails with a TypeError

**Environment Information**
```
angr environment report
=============================
Date: 2019-02-23 19:44:25.549336
Running in virtual environment at /home/lockshaw/.virtualenvs/broken3
Platform: linux-x86_64
Python version: 3.7.2 (default, Jan 10 2019, 23:51:51) 
[GCC 8.2.1 20181127]
######## angr #########
Python found it in /home/lockshaw/hax/dev/angr/broken/angr/angr
Pip version angr 8.19.2.4
Git info:
        Current commit 956b75c1908b786d51a5af3a29a96b5ef35c756e from branch master
        Checked out from remote github: git@github.com:angr/angr.git
######## ailment #########
Python found it in /home/lockshaw/hax/dev/angr/broken/ailment/ailment
Pip version ailment 8.19.2.4
Git info:
        Current commit bea5c50e987406af0264922cf8f10a765fdda210 from branch master
        Checked out from remote github: git@github.com:angr/ailment.git
######## cle #########
Python found it in /home/lockshaw/hax/dev/angr/broken/cle/cle
Pip version cle 8.19.2.4
Git info:
        Current commit 89c84c1ea9ee6cd29b37c5d7359cf39a0e15dfde from branch master
        Checked out from remote github: git@github.com:angr/cle.git
######## pyvex #########
Python found it in /home/lockshaw/hax/dev/angr/broken/pyvex/pyvex
Pip version pyvex 8.19.2.4
Git info:
        Current commit 050ffe4f45deb5e1aaad37312d83972c907682c3 from branch master
        Checked out from remote github: git@github.com:angr/pyvex.git
######## claripy #########
Python found it in /home/lockshaw/hax/dev/angr/broken/claripy/claripy
Pip version claripy 8.19.2.4
Git info:
        Current commit ff1571cc1b9079d43d9e7004da230c25b740f9f5 from branch master
        Checked out from remote github: git@github.com:angr/claripy.git
######## archinfo #########
Python found it in /home/lockshaw/hax/dev/angr/broken/archinfo/archinfo
Pip version archinfo 8.19.2.4
Git info:
        Current commit 0019bbb37ebd0ee24fc4b1c637e5d097add9d326 from branch master
        Checked out from remote github: git@github.com:angr/archinfo.git
######## ana #########
Python found it in /home/lockshaw/hax/dev/angr/broken/ana/ana
Pip version ana 0.5
Git info:
        Current commit 48657dc6500e33f5818b98003f1f179bfe03e1cc from branch master
        Checked out from remote github: git@github.com:zardus/ana.git
######## z3 #########
Python found it in /home/lockshaw/.virtualenvs/broken3/lib/python3.7/site-packages/z3
Pip version not found!
Couldn't find git info
######## unicorn #########
Python found it in /home/lockshaw/.virtualenvs/broken3/lib/python3.7/site-packages/unicorn
Pip version unicorn 1.0.1
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/lockshaw/hax/dev/angr/broken/angr/angr/lib/angr_native.so', handle 555e80fb5c90 at 0x7f84aa4e5be0>
unicorn: <CDLL '/home/lockshaw/.virtualenvs/broken3/lib/python3.7/site-packages/unicorn/lib/libunicorn.so', handle 555e80a8e590 at 0x7f84a976a978>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f84aa419cf8>
z3: <CDLL '/home/lockshaw/.virtualenvs/broken3/lib/python3.7/site-packages/z3/lib/libz3.so', handle 555e817e23b0 at 0x7f84a3e16ba8>
```

**To Reproduce**
```
import angr

p = angr.Project('ls', auto_load_libs=False)
cfg = p.analyses.CFGFast()
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-34d0f86d8755> in <module>
----> 1 cfg = p.analyses.CFGFast()

~/hax/dev/angr/broken/angr/angr/analyses/analysis.py in __call__(self, *args, **kwargs)
    107 
    108         oself._show_progressbar = show_progressbar
--> 109         oself.__init__(*args, **kwargs)
    110         return oself
    111 

~/hax/dev/angr/broken/angr/angr/analyses/cfg/cfg_fast.py in __init__(self, binary, objects, regions, pickle_intermediate_results, symbols, function_prologues, resolve_indirect_jumps, force_segment, force_complete_scan, indirect_jump_target_limit, collect_data_references, extra_cross_references, normalize, start_at_entry, function_starts, extra_memory_regions, data_type_guessing_handlers, arch_options, indirect_jump_resolvers, base_state, exclude_sparse_regions, skip_specific_regions, heuristic_plt_resolving, detect_tail_calls, low_priority, cfb, start, end, **extra_arch_options)
   1071 
   1072         # Start working!
-> 1073         self._analyze()
   1074 
   1075     def __getstate__(self):

~/hax/dev/angr/broken/angr/angr/analyses/forward_analysis.py in _analyze(self)
    588             self._analysis_core_graph()
    589 
--> 590         self._post_analysis()
    591 
    592     def _analysis_core_graph(self):

~/hax/dev/angr/broken/angr/angr/analyses/cfg/cfg_fast.py in _post_analysis(self)
   1573 
   1574         if self.project.arch.name in ('X86', 'AMD64', 'MIPS32'):
-> 1575             self._remove_redundant_overlapping_blocks()
   1576 
   1577         self._updated_nonreturning_functions = set()

~/hax/dev/angr/broken/angr/angr/analyses/cfg/cfg_fast.py in _remove_redundant_overlapping_blocks(self)
   2837                             nop_length = self._get_nop_length(insns)
   2838 
-> 2839                     if nop_length <= 0:
   2840                         continue
   2841 

TypeError: '<=' not supported between instances of 'NoneType' and 'int'
```

**Additional context**
[ls.zip](https://github.com/angr/angr/files/2897558/ls.zip)

**Comment by rupran** [comment]
I ran into the same problem with `curl` from openWRT which is super-stripped (no section header etc.) so I first thought this was the problem but the attached binary looks like a regular ELF file.

In my case `nop_length` stays `None` because we end up in the capstone case and `insns` is the empty list. This means the check `if insns:` will be False and the assignment in line 2837 (in the stacktrace above) is never executed.

Should we simply initialize `nop_length` to `0` instead of `None`?

**Comment by lockshaw** [comment]
Appears to be fixed now!

**Comment by ltfish** [comment]
Addressed by this commit: e0689f1d.
